src/Shared/PropertyHelper/PropertyHelper.cs

```diff
@@ -151,6 +151,7 @@ public void SetValue(object instance, object? value)
     /// A cached array of all public properties of the specified type.
     /// </returns>
     [RequiresUnreferencedCode("This API is not trim safe.")]
+    [RequiresDynamicCode("This API is not trim safe.")]
```

Comment:
Could you move `"This API is not trim safe."` to a const on this type and reuse in all the attributes. Some are inconsistent.